lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -73,29 +73,46 @@ defmodule Thrift.Generator.StructBinaryProtocol do
 
     struct_matcher = {:%, [], [name, {:%{}, [], field_matchers}]}
 
-    field_serializers = Enum.map(fields, fn %Field{name: name, type: type, id: id} ->
-      var = Macro.var(name, nil)
-      quote do
-        case unquote(var) do
-          nil ->
-            <<>>
-          _ ->
-            unquote([
-              quote do <<unquote(type_id(type, file_group)), unquote(id) :: size(16)>> end,
-              value_serializer(type, var, file_group)
-            ] |> Utils.merge_binaries |> Utils.simplify_iolist)
+    reject_void_fields = fn %Field{type: type} -> type != :void end
+
+    field_serializers = Enum.filter_map(fields, reject_void_fields,
+      fn %Field{name: name, type: type, id: id} ->
```

Comment:
That doesn't work though, it still creates a serializer and deserializer for them (and we get complaints about unused variables), we don't want that. When we get a void field, we just want it totally skipped.